### PR TITLE
switch services to thiserror

### PR DESCRIPTION
**Pull request by ctaggart**
See https://github.com/ctaggart/autorust/pull/92. This pull request switches the services from snafu to thiserror. It compiles. The error descriptions need some improvement. Overall impression of `thiserror` is good.

### CODE REVIEW

**Comment by heaths** [comment]
977 files is a lot to review. I realized most of these are generated. In our other language repos, we always generate to a specific subfolder (e.g. "generated") and that is configured as a generated folder, which GitHub then hides such files by default (makes it easier to filter and review actual person-effected changes). Would that make sense here as well? Could you generate code to a "generated" folder and re-export* as needed?

\* For now. As we discussed in our meeting, eventually we probably want all generated code internal and expose what we need from hand-crafted clients - often just thin facades on generated code, but not always - like Key Vault.

**Comment by ctaggart** [comment]
All files in this pull request are generated. I'll look into marking the [files as generated](https://medium.com/@clarkbw/managing-generated-files-in-github-1f1989c09dfd). That is a good idea. I would recommend just picking a file or two as a sample.

**Comment by heaths** [comment]
What I recommend is: services/mgmt/addons/src/package_2017_05/operations.rs -> services/mgmt/addons/src/generated/package_2017_05/operations.rs

All our language SDKs do something like this, and then import/use generated classes as needed. In the Rust SDK case, we can choose to re-export those publicly (we do this for "low-level SDKs", which are just generated and published) or import them into custom clients as needed (for our "hand-crafted SDKs", where we do some massaging of structures or value-add features) based on the clients' service version the user specified.

This also makes it easy to mark them as generated since it fits a pattern. Example for .NET: https://github.com/Azure/azure-sdk-for-net/blob/be7acee5bf7432a09daa3f06568e04a5d31e5059/.gitattributes#L20-L25

File: services/mgmt/addons/src/package_2017_05/operations.rs [hunk 1]
```diff
@@ -93,71 +90,73 @@ pub mod support_plan_types {
...
-                source: Box<dyn std::error::Error + Sync + Send>,
-            },
-            DeserializeError {
-                source: serde_json::Error,
-                body: bytes::Bytes,
-            },
-            GetTokenError {
-                source: azure_core::errors::AzureError,
-            },
+            #[error("ParseUrlError")]
```
Comment (heaths) [inline services/mgmt/addons/src/package_2017_05/operations.rs:148]: A lot of duplication, even though generated (larger crates). Is it worth consolidating, or is this basically a base "template" (common errors) + any specifics documented in swagger not in the 2xx range? By consolidation, I mean: why not define them once per library/service version?

Comment (ctaggart) [reply services/mgmt/addons/src/package_2017_05/operations.rs:148]: Each API operation defines its own responses. Yes, there is a set of common errors and anything not in the 2xx range like you guessed.

**Comment by ctaggart** [comment]
I'm going to close this pull request. I'll open a new one with updated services.

**Pull request closed without merging**